### PR TITLE
WIP - Importer

### PR DESCRIPTION
**Pull request by thijscobben**
This code needs some restructuring and a thorough review :-)
It imports a lot of records and (large) files from b2share_v1 though.
See script in b2share_demo/re-import.sh how to clean db and re-import.
--
Known incompletes/issues:
This script is to import a lot of data from v1 to populate your instance for testing. It does not (yet) import all data from v1.
- domain_metadata should be mapped to community specific schemas
- an error report should be created listing exactly which records and files are not imported and for what reason
- there is a request to add checksum checking on each downloaded file

In general, the use of current_app.logger has been a bit cluttery.

### CODE REVIEW

File: b2share/modules/communities/api.py [hunk 1]
```diff
@@ -89,15 +89,26 @@ def get(cls, id=None, name=None, with_deleted=False):
...
+                metadata = CommunityMeta.query.order_by(CommunityMeta.created)
+            else:
+                metadata =  CommunityMeta.query.filter(
+                    CommunityMeta.name.like(name)
+                    ).order_by(CommunityMeta.created)
         elif not(start is None) and not(stop is None):
-            metadata = CommunityMeta.query.order_by(CommunityMeta.created).limit(stop)[start:]
+            if name is None:
+                metadata = CommunityMeta.query.order_by(
+                    CommunityMeta.created).order_by(
```
Comment (emanueldima) [inline b2share/modules/communities/api.py]: two times order_by?

File: b2share/modules/schemas/helpers.py [hunk 2]
```diff
@@ -160,3 +160,6 @@ def load_root_schemas(cli=False, verbose=False):
                 raise
     if cli:
         click.secho('LOADED {} schemas'.format(configs_count), fg='green')
+
```
Comment (emanueldima) [inline b2share/modules/schemas/helpers.py]: this should not be committed

File: demo/b2share_demo/cli.py [hunk 3]
```diff
@@ -93,3 +93,43 @@ def load_config(verbose, force):
...
+@click.argument('token')
+@click.argument('download_directory')
+def import_v1_data(verbose, download, token,
+         download_directory,limit):
+    if verbose:
+        click.secho("Importing data from b2share.eudat.eu to this instance")
+        os.chdir(download_directory)
+        if download:
+            if verbose:
+                click.secho("Cleaning download directory %s" % download_directory)
```
Comment (emanueldima) [inline demo/b2share_demo/cli.py]: Removing files like this is potentially dangerous. Isn't it better to just quit if the directory is not empty?

File: demo/b2share_demo/helpers.py [hunk 4]
```diff
@@ -269,3 +273,135 @@ def community_id_match(match):
...
+    params['access_token'] = token
+    params['page_size']=100
+    params['page_offset']=0
+    finished = False
+    counter = 0
+    if not(limit is None) and counter*100 + 100 > limit:
+        params['page_size']= limit
+    while not finished:
+        r = requests.get(url, params=params, verify=False)
+        f = open(target_dir+ "/%d.txt" % counter,'w')
```
Comment (emanueldima) [inline demo/b2share_demo/helpers.py]: os.path.join is better

File: demo/b2share_demo/helpers.py [hunk 5]
```diff
@@ -269,3 +273,135 @@ def community_id_match(match):
...
+    #assume the current community specific block schema for metadata format
+    return result
+    
+def _process_files(deposit, r, download, download_dir):
+    if not(r['files']=='RESTRICTED'):
+        for file_dict in r['files']:
+            if download:
+                try:
+                    if 'name' in file_dict.keys():
+                        #only for gangsters - bypass HTTPS verification see
```
Comment (emanueldima) [inline demo/b2share_demo/helpers.py]: Where is the certificate validation bypassed? I don't see it here

Comment (thijscobben) [inline demo/b2share_demo/helpers.py]: It was moved to _save_file I didn't move the comment obviously. FIXED.

**Comment by emanueldima** [comment]
closing this one, as the code has been merged in a different PR:  https://github.com/EUDAT-B2SHARE/b2share/pull/1131

**Pull request closed without merging**